cmd/k8s-proxy/internal/config/config.go

```diff
@@ -204,7 +213,24 @@ func (l *configLoader) watchConfigSecretChanges(ctx context.Context, secretName
 		select {
 		case <-ctx.Done():
 			return ctx.Err()
-		case ev := <-w.ResultChan():
+		case ev, ok := <-w.ResultChan():
+			if !ok {
+				w.Stop()
+				w, err = secrets.Watch(ctx, metav1.ListOptions{
```

Comment:
this feels reasonable to me. "If we fail to read from the result channel, stop the watcher, try to reestablish, and if that fails, error out".